packages/xrpl/test/integration/utils.ts

```diff
@@ -1,45 +1,197 @@
+ * By default throttle doesn't return a promise for async functions unless it's invoking them immediately.
+ * See CUR-4769 for details.
+ *
+ * @param func - async function to throttle calls for.
+ * @param wait - same function as lodash.throttle's wait parameter. Call this function at most this often.
+ * @returns a promise which will be resolved/ rejected only if the function is executed, with the result of the underlying call.
+ */
+function asyncThrottle<F extends (...args: unknown[]) => Promise<unknown>>(
+  func: F,
+  wait?: number,
+): (...args: Parameters<F>) => ReturnType<F> {
+  const throttled = throttle((resolve, reject, args: Parameters<F>) => {
+    func(...args)
+      .then(resolve)
+      .catch(reject)
+  }, wait)
+  const ret = (...args: Parameters<F>): ReturnType<F> =>
+    new Promise((resolve, reject) => {
+      throttled(resolve, reject, args)
+    }) as ReturnType<F>
+  return ret
+}
+
+const throttledLedgerAccept = asyncThrottle(sendLedgerAccept, 1000)
+
+export async function ledgerAccept(
+  client: Client,
+  retries?: number,
+  shouldThrottle?: boolean,
+): Promise<unknown> {
+  return new Promise<unknown>((resolve, reject) => {
+    const ledgerAcceptFunc = shouldThrottle
+      ? throttledLedgerAccept
+      : sendLedgerAccept
+    ledgerAcceptFunc(client)
+      .then(resolve)
+      .catch((error) => {
+        if (retries === undefined) {
+          setTimeout(() => {
+            resolve(ledgerAccept(client, 10))
+          }, 1000)
+        } else if (retries > 0) {
+          setTimeout(() => {
+            resolve(ledgerAccept(client, retries - 1))
+          }, 1000)
+        } else {
+          reject(error)
+        }
+      })
+  })
+}
+
+/**
+ * Attempt to get the time after which we can check for the escrow to be finished.
+ * Sometimes the ledger close_time is in the future, so we need to wait for it to catch up.
+ *
+ * @param targetTime - The target wait time, before accounting for current ledger time.
+ * @param minimumWaitTimeInMs - The minimum wait time in milliseconds.
+ * @param maximumWaitTimeInMs - The maximum wait time in milliseconds.
+ * @returns The wait time in milliseconds.
+ */
+export function calculateWaitTimeForTransaction(
+  targetTime: number,
+  minimumWaitTimeInMs = 5000,
+  maximumWaitTimeInMs = 20000,
+): number {
+  const currentTimeUnixInMs = Math.floor(new Date().getTime())
+  const currentTimeRippleInSeconds = unixTimeToRippleTime(currentTimeUnixInMs)
+  const closeTimeCurrentTimeDiffInSeconds =
+    currentTimeRippleInSeconds - targetTime
+  const closeTimeCurrentTimeDiffInMs = closeTimeCurrentTimeDiffInSeconds * 1000
+  return Math.max(
+    minimumWaitTimeInMs,
+    Math.min(
+      Math.abs(closeTimeCurrentTimeDiffInMs) + minimumWaitTimeInMs,
+      // Maximum wait time of 20 seconds
+      maximumWaitTimeInMs,
+    ),
+  )
+}
+
+export function subscribeDone(client: Client): void {
   client.removeAllListeners()
-  done()
+}
+
+export async function runCommand({
```

Comment:
Commands are generally `requests`, not `transactions`, so i'd suggest renaming this if it's just for Transactions.

```suggestion
export async function submitTransaction({
```

Reply:
Unresolving this because I was thoroughly confused why a method named `runCommand` wasn't taking a request as a parameter, and was instead working with transactions and wallets.

Reply:
Changed